test-series/4/tools/sim_prep.py

```diff
@@ -0,0 +1,626 @@
+            [np.allclose(pos, ref_pos, atol=tol) for ref_pos in ref_antpos.values()]
+        ).flatten()
+        if refant_index.size == 0:
+            continue
+        antenna_map[ant] = refants[refant_index[0]]
+    return antenna_map
+
+def _get_antpos(uvd, ENU=True):
+    """Retrieve the {ant : pos} dictionary from the data."""
+    if ENU:
+        pos, ant = uvd.get_ENU_antpos()
+    else:
+        ant = uvd.antenna_numbers
+        pos = uvd.antenna_positions
+    return dict(zip(ant, pos))
+
+def _get_optimal_translation(sim_antpos, ref_antpos, tol=1.0):
+    """Find the translation that maximizes the overlap between antenna arrays."""
+    # Get a dictionary of translations; keys are sim_ant -> ref_ant.
+    translations = _build_translations(sim_antpos, ref_antpos, tol)
+    intersection_sizes = {}
+    
+    # Calculate the number of antennas in the intersection for each pair of arrays.
+    for sim_ant_to_ref_ant, translation in translations.items():
+        new_sim_antpos = {
+            ant : pos + translation for ant, pos in sim_antpos.items()
+        }
+        Nintersections = 0
+        for pos in new_sim_antpos.values():
+            pos_in_ref = any(
+                [np.allclose(pos, ref_pos, atol=tol)
+                 for ref_pos in ref_antpos.values()]
+            )
+            Nintersections += pos_in_ref
+        intersection_sizes[sim_ant_to_ref_ant] = Nintersections
+        
+    # Choose the translation that has the most antennas in the intersection.
+    sim_to_ref_keys = list(translations.keys())
+    intersections_per_translation = np.array(list(intersection_sizes.values()))
+    index = np.argmax(intersections_per_translation)
+    return translations[sim_to_ref_keys[index]]
+
+def _build_translations(sim_antpos, ref_antpos, tol=1.0):
+    """Build all possible translations that overlap at least one antenna."""
+    sim_ants = list(sim_antpos.keys())
+    ref_ants = list(ref_antpos.keys())
+    translations = {
+        f"{sim_ant}->{ref_ant}" : ref_antpos[ref_ant] - sim_antpos[sim_ant]
+        for sim_ant in sim_ants for ref_ant in ref_ants
+    }
+    unique_translations = {}
+    for key, translation in translations.items():
+        if not unique_translations:
+            unique_translations[key] = translation
+            continue
+        keep_translation = not any(
+            [np.allclose(translation, unique_translation, atol=tol)
+             for unique_translation in unique_translations.values()]
+        )
+        if keep_translation:
+            unique_translations[key] = translation
+    return unique_translations
+
+def _parse_filename_for_cmp(filename):
+    """Infer the sky component from the provided filename."""
+    cmp_re = re.compile("[a-z]+.uvh5")
+    sky_cmp = cmp_re.findall(str(filename))
+    if sky_cmp == []:
+        raise ValueError(
+            f"Simulation component could not be inferred from {filename}."
+        )
+    return sky_cmp[0][:-5]
```

Comment:
I'll check it out and see if it's worth pulling into here. The idea behind this function was that it would just be acting on one of the following files: `foregrounds.uvh5`, `eor.uvh5`, `sum.uvh5` (these are the compressed data sets for the full day of simulated data)

Reply:
Ahhh OK. Yeah I don't think the notebook function is for those files, it's for the chunked ones.